pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/AbstractApexUnitTestRule.java

```diff
@@ -31,12 +31,9 @@ public AbstractApexUnitTestRule() {
      */
     @Override
     public Object visit(final ASTUserClass node, final Object data) {
-        /*
-        if (!isTestMethodOrClass(node) && node.getApexVersion() >= Version.V176.getExternal()) {
+        if (!isTestMethodOrClass(node) /*&& node.getApexVersion() >= Version.V176.getExternal()*/) { // TODO(b/243906862)
```

Comment:
That is also unclear to me.

Right, the only uses "ApexVersion" are to enable test-only rules on test code that predates `@IsTest` introduces in API version 24.0, circa 2012. In the "worst case", assuming that Jorje actually does read the metadata XML, then the impact of not reimplementing this would be false-negatives for test-only rules on pre-2012 code. If it doesn't, then there is no impact.

I created https://github.com/pmd/pmd/pull/4280 to deprecate `ApexRootNode.getApexVersion` in master and will update this PR shortly.